bakery/checkers/checkers.go

```diff
@@ -4,192 +4,187 @@ package checkers
 
 import (
 	"fmt"
-	"net"
+	"sort"
 	"strings"
 
+	"golang.org/x/net/context"
 	"gopkg.in/errgo.v1"
 )
 
+const StdNamespace = "std"
+
 // Constants for all the standard caveat conditions.
 // First and third party caveat conditions are both defined here,
 // even though notionally they exist in separate name spaces.
 const (
-	CondDeclared     = "declared"
-	CondTimeBefore   = "time-before"
-	CondClientIPAddr = "client-ip-addr"
-	CondClientOrigin = "origin"
-	CondError        = "error"
-	CondNeedDeclared = "need-declared"
-	CondAllow        = "allow"
-	CondDeny         = "deny"
+	CondDeclared   = "declared"
+	CondTimeBefore = "time-before"
+	CondError      = "error"
+	CondAllow      = "allow"
+	CondDeny       = "deny"
 )
 
-// ErrCaveatNotRecognized is the cause of errors returned
-// from caveat checkers when the caveat was not
-// recognized.
-var ErrCaveatNotRecognized = errgo.New("caveat not recognized")
-
-// Caveat represents a condition that must be true for a check to
-// complete successfully. If Location is non-empty, the caveat must be
-// discharged by a third party at the given location.
-// This differs from macaroon.Caveat in that the condition
-// is not encrypted.
-type Caveat struct {
-	Location  string
-	Condition string
-}
-
-// Checker is implemented by types that can check caveats.
-type Checker interface {
-	// Condition returns the identifier of the condition
-	// to be checked - the Check method will be used
-	// to check caveats with this identifier.
-	//
-	// It may return an empty string, in which case
-	// it will be used to check any condition
-	Condition() string
-
-	// Check checks that the given caveat holds true.
-	// The condition and arg are as returned
-	// from ParseCaveat.
-	//
-	// For a checker with an empty condition, a
-	// return of bakery.ErrCaveatNotRecognised from
-	// this method indicates that the condition was
-	// not recognized.
-	Check(cond, arg string) error
-}
+const (
+	CondNeedDeclared = "need-declared"
```

Comment:
'cos it's a third party caveat, not a first party one. It will end up going probably, as I think third party caveats will change format quite a but.